presto-main/src/main/java/io/prestosql/operator/JoinUtils.java

```diff
@@ -65,6 +67,16 @@ public static boolean isBuildSideReplicated(JoinNode joinNode)
                 .matches();
     }
 
+    public static boolean isBuildSideReplicated(SemiJoinNode semiJoinNode)
```

Comment:
Please add `isBuildSideReplicated(PlanNode node)`. Then we wouldn't need to do ifs in `DynamicFilterService`